### ISSUE TITLE
A11y_.NET Core_WPF_ExpenseItIntro_ViewExpenseReport_Contrast: The color contrast for "Names" list item is less than 4.5:1.

### ISSUE THREAD

**Issue by Swati1700** [dotnet/wpf#6736]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags**:#A11yMAS;#A11yTCS;#.NET Core[WPF]-Win32-June2022;#.NET Core;#A11ySev2;#WCAG1.4.3;#DesktopApp;#Win11;#ColorContrast;#Benchmark;

### Environment Details:
.NET Core
WPF
Operating System: Windows 11 Enterprise 21H2

### Steps to Reproduce:

1. Launch VS 2019 Int Preview
2. Navigate to Getting started and right click on the Expense it intro and click on build
3. Then right click on Expense it Intro then Click on Debug-->Start New Instance. 
4. Verify all the elements of 'View Expense Report' 
5. Turn on Color contrast analyzer and check that The Color contrast ratio for "Names" list item is less than 4.5:1.

### Actual:

The color contrast for "Names" list item is less than 4.5:1.

**Similar issue observed in the below scenario:**
**Issue: ExpenseItIntro - Expense Report for**

1. Launch VS 2019 Int Preview
2. Navigate to Getting started and right click on the Expense it intro and click on build
3. Then right click on Expense it Intro then Click on Debug-->Start New Instance. 
4. Navigate to Names list and select any name and click on View.
5. Verify all the elements of 'Expense Report for' screen 
6. Turn on Color contrast analyzer and check that The Color contrast ratio for "ExpenseType" and "Amount" and " is less than 4.5:1.

### Expected:
The color contrast for "Names" list item should be more than or equal to 4.5:1.

### User Impact:
Low vision users won't be able to see the list item.

![Screenshot (1160)](https://user-images.githubusercontent.com/86710888/176378794-a89da405-9171-4c64-a297-d7ff64f50cbe.png)

**Comment by merriemcgaw** [comment]
@anjalisheel-wpf - I think you are looking at a different color than the tester is. What has to be 4.5:1 is the word "Name" in the blue header box. Can you please check that and make sure that it is pure white? I think it is ever so slightly off white and that is the cause of the contrast problems.

**Comment by anjalisheel-wpf** [comment]
@merriemcgaw Check the attached video, now I am referring to the correct label. The font is small, thats why we have to scroll a little in order to get the desired ratio.

https://user-images.githubusercontent.com/85492722/183872116-a096f97d-ec3d-44bc-a658-ec0211aebee3.mov

**Comment by merriemcgaw** [comment]
I stand corrected 😺 . @Swati1700 can you please take a look at this? It seems good to me.

**Comment by anjalisheel-wpf** [comment]
The code which we are using for the name header - 

<Border Grid.Column="1" Grid.Row="1" Style="{StaticResource ListHeaderStyle}">
    <Label Style="{StaticResource ListHeaderTextStyle}">Names</Label>
</Border>
        
        
And the associated resource file code is - 

 
    <Style x:Key="ListHeaderTextStyle" TargetType="{x:Type Label}">
        <Setter Property="Foreground" Value="White" />
        <Setter Property="VerticalAlignment" Value="Center" />
        <Setter Property="HorizontalAlignment" Value="Left" />
        <Style.Triggers>
            
            <DataTrigger Binding="{Binding Path=(SystemParameters.HighContrast)}" Value="true">
                <Setter Property="Foreground" Value="{DynamicResource {x:Static SystemColors.ControlTextBrushKey}}"/>
            </DataTrigger>
        </Style.Triggers>
    </Style>

**Comment by Swati1700** [comment]
Verified the bug in the environment and issue is still reproducing.

https://user-images.githubusercontent.com/86710888/184115397-77f0ffb6-650f-4569-a606-ebdd4f03c22d.mp4

**Comment by anjalisheel-wpf** [comment]
@Swati1700 You need to be very precise with the color picker, because text font size is small, test it with little precision.

**Comment by pchaurasia14** [comment]
@Swati1700  - If you look at the code snippet shared by @anjalisheel-wpf, we are explicitly only setting foreground to `White` and the highlight color is set to whatever system default is.

**Comment by Swati1700** [comment]
We have verified the bug on the below two environment and the issue is still reproducing.
Microsoft Visual Studio Enterprise 2022 (64-bit) - Int Preview
Version 17.4.0 Preview 2.0 [32731.56.main] and Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.3.0 Preview 5.0. 
Could you please confirm on which environment version you are verifying the fixes so that we can try it on our end.

**Comment by merriemcgaw** [comment]
@pchaurasia14 I think the best approach is to update the colors here. If the white text over the system highlight color doesn't produce the right contrast ratio, perhaps you could also use system colors like you have in the case of high contrast? Another option is to change the background color to whatever you would like it to be, it doesn't need to be a system default color does it?

**Comment by merriemcgaw** [comment]
@Swati1700 this one is ready for your review.

**Comment by Swati1700** [comment]
We have verified the bug on the below two environment
Microsoft Visual Studio Enterprise 2022 (64-bit) - Int Preview
Version 17.4.0 Preview 2.0 [32731.56.main] and Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.3.0 Preview 5.0.
We have changed the background color and checked the color contrast ratio, but it is less than 4.5:1.

**Comment by anjalisheel-wpf** [comment]
@Swati1700 Check recorded video and snapshot attached. I pushed changes, so take latest and then try.
<img width="914" alt="Screenshot 2022-08-17 182542" src="https://user-images.githubusercontent.com/85492722/185139598-37e307cf-8803-45f3-92b1-0812d605a65e.png">

https://user-images.githubusercontent.com/85492722/185139811-c735812f-db74-4f0b-8716-e191cf6ef537.mov

**Comment by merriemcgaw** [comment]
@Swati1700 it looks like you need to download the latest samples from the samples repo to see this fix.

**Comment by Swati1700** [comment]
Verified the bug in the below environment and the bug is fixed.
Microsoft Visual Studio Enterprise 2022 (64-bit) - Int Preview
Version 17.4.0 Preview 2.0 [32814.64.main]
Bug does not repro. Hence closing the bug.
![Screenshot (1318)](https://user-images.githubusercontent.com/86710888/185326435-fdebbaed-6054-4e46-b70a-d1ca10f7221f.png)
